app/src/main/java/ie/macinnes/tvheadend/player/TvheadendPlayer.java

```diff
@@ -235,46 +235,11 @@ public void seek(long timeMs) {
     public void setPlaybackParams(PlaybackParams params) {
         float rawSpeed = params.getSpeed();
         int speed = (int) rawSpeed;
-        int translatedSpeed;
-
-        switch(speed) {
-            case 0:
-                translatedSpeed = 100;
-                break;
-            case -2:
-                translatedSpeed = -200;
-                break;
-            case -4:
-                translatedSpeed = -300;
-                break;
-            case -12:
-                translatedSpeed = -400;
-                break;
-            case -48:
-                translatedSpeed = -500;
-                break;
-            case 2:
-                translatedSpeed = 200;
-                break;
-            case 4:
-                translatedSpeed = 300;
-                break;
-            case 12:
-                translatedSpeed = 400;
-                break;
-            case 48:
-                translatedSpeed = 500;
-                break;
-            default:
-                Log.d(TAG, "Unknown speed??? " + rawSpeed);
-            return;
-        }
-
-        Log.d(TAG, "Speed: " + params.getSpeed() + " / " + translatedSpeed);
 
+        Log.d(TAG, "Speed: " + params.getSpeed());
         if (mDataSource != null) {
-            mDataSource.setSpeed(translatedSpeed);
-            mExoPlayer.setPlaybackParameters(new PlaybackParameters(translatedSpeed, 0));
+            mDataSource.setSpeed(speed);
+            mExoPlayer.setPlaybackParameters(new PlaybackParameters(speed, 1));
```

Comment:
Are you sure this works out? 

`mDataSource.setSpeed(speed);` passes the speed value over to TVheadend, which expects:

    speed              u32   required   Specify speed (0=pause, 100=1x fwd, -100=1x backward)

http://tvheadend.org/projects/tvheadend/wiki/Htsp#subscriptionSpeed-Added-in-version-9

Reply:
Or, maybe it doesn't get passed over! Apparently, I haven't committed that code ;)

https://github.com/kiall/android-tvheadend/blob/develop/app/src/main/java/ie/macinnes/tvheadend/player/HtspSubscriptionDataSource.java#L330

Reply:
Re-added the translated speed in ed1dcf9 and 3284969